src/Reflection/ReflectionClass.php

```diff
@@ -242,7 +242,15 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $name = 'class@anonymous';
+        if ($this->getFileName() !== null) {
+            $name .= ' ' . $this->getFileName();
```

Comment:
I would at least add the line number. Also, spaces are not really usable in class names, IIRC. The typical naming is something like `class@anonymous/in/6UEGQ0x7f5b24d0e025`